src/inventorymanager.cpp

```diff
@@ -440,25 +442,20 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 			src_can_take_count = dst_can_put_count = 0;
 	}
 
-	int old_count = count;
+	int old_move_count = move_count;
 
-	/* Modify count according to collected data */
-	count = src_item.count;
-	if (src_can_take_count != -1 && count > src_can_take_count)
-		count = src_can_take_count;
-	if (dst_can_put_count != -1 && count > dst_can_put_count)
-		count = dst_can_put_count;
+	// Apply limits given by allow_* callbacks
+	if (src_can_take_count != -1 && (int)move_count > src_can_take_count)
+		move_count = src_can_take_count;
+	if (dst_can_put_count != -1 && (int)move_count > dst_can_put_count)
+		move_count = dst_can_put_count;
 
-	/* Limit according to source item count */
-	if (count > list_from->getItem(from_i).count)
-		count = list_from->getItem(from_i).count;
+	// Limit just in case
+	if (move_count > list_from->getItem(from_i).count)
+		move_count = list_from->getItem(from_i).count;
```

Comment:
not 100% sure I'd rather convert this to an assert
(should not happen at all looking at the rest of the code)

Reply:
I'll instead reword the comment above to clarify that the ItemStack might still differ due to allow_* callback execution, even though mods should not use that to modify the inventory.